src/api/resources/resources.ts

```diff
@@ -347,9 +360,33 @@ const FROM_POST_MAPPERS: {
         }
       }
     }
-    name && (result.name = name)
-    description && (result.description = description)
-    properties && (result.feature.properties = properties)
+    name && (result.name = name) && (properties.name = name)
+    description &&
+      (result.description = description) &&
+      (properties.description = description)
+    result.feature.properties = properties
+    return result
+  },
+  routes: (data: any) => {
+    const { name, description, points, properties = {} } = data
+    const distance = calculateDistance(points)
+    const result: Route = {
+      feature: {
+        type: 'Feature',
+        geometry: {
+          type: 'LineString',
+          coordinates: points.map((p: Position) => {
+            return [p.longitude, p.latitude]
+          })
+        }
+      },
+      distance
+    }
+    name && (result.name = name) && (properties.name = name)
```

Comment:
Adding name & description to properties is a good idea.... Freeboard-SK already does this to support various functions.

Reply:
I think this was already in your buildXx code or someplace.